lib/tbd/psi.rb

```diff
@@ -877,10 +877,13 @@ def derate(id = "", s = {}, lc = nil)
       return hashkey(tag, s, k, mth, ERR) unless s.key?(k)
 
       case k
-      when :heatloss, :net, :r
+      when :heatloss
+        return mismatch(tag, s[k], Numeric, mth) unless s[k].respond_to?(:to_f)
+        return zero(tag, mth, WRN)                   if s[k].to_f.abs < 0.001
```

Comment:
An unfortunate typo, snuck in when refactoring checks. Negative heat loss (e.g. corners) is admissible, while 0 W/K.m triggers a warning. Without the fix (`.abs`), any negative heat loss logged a "zero heat loss" warning.